examples/run_setup_tutorial.py

```diff
@@ -0,0 +1,100 @@
+"""
+=========
+Run Setup
+=========
+
+By: Jan N. van Rijn
+
+One of the key features of the openml-python library is that is allows to
+reinstantiate flows with hyperparameter settings that were uploaded before.
+This tutorial uses the concept of setups. Although setups are not extensively
+described in the OpenML documentation (because most users will not directly
+use them)
+
+A key requirement for reinstantiating a flow is to have the same scikit-learn
+version as the flow that was uploaded. This tutorial will upload the flow
+itself, so it can be ran with any scikit-learn version that is supported by
+this library.
+
+In this tutotial we will
+    1) Create a flow and use it to solve a task
+    2) Download the flow, reinstantiate the model with same hyperparameters,
+       and solve the same task again.
+    3) We will verify that the obtained results are exactly the same.
+Readers interested in reinstantiating a setup can skip part 1 and 2 and start
+with part 3 immediately.
+"""
+import logging
+import numpy as np
+import openml
+import sklearn.ensemble
+import sklearn.impute
+import sklearn.preprocessing
+
+
+root = logging.getLogger()
+root.setLevel(logging.INFO)
+
+###############################################################################
+# 1) Create a flow and use it to solve a task
+###############################################################################
+
+# first, let's download the task that we are interested in
+task = openml.tasks.get_task(6)  # letter dataset
+
+
+# we will create a fairly complex model, with many preprocessing components and
+# many potential hyperparameters. Of course, the model can be as complex and as
+# easy as you want it to be
+model_original = sklearn.pipeline.make_pipeline(
+    sklearn.impute.SimpleImputer(),
+    sklearn.preprocessing.StandardScaler(),
+    sklearn.ensemble.RandomForestClassifier()
+)
+
+
+# Let's change some hyperparameters. Of course, in any good application we
+# would tune them using, e.g., Random Search or SMAC, but for the purpose of
+# this tutorial we set them to some specific values that might or might not be
+# optimal
+hyperparameters_original = {
+    'simpleimputer__strategy': 'median',
+    'randomforestclassifier__random_state': 42,
+    'randomforestclassifier__min_samples_leaf': 1,
```

Comment:
That's already the scikit-learn default, could you please change it to something else?

Reply:
However, 1 is (empirically) the optimal value, see: 
https://arxiv.org/abs/1710.04725

I would prefer to keep it to 1, and rather add another hyperparameter to ensure enough diversity from the default.